### PR TITLE
BACK-1789 : DD context logging

### PR DESCRIPTION
**Pull request by gagbo**
Create a Datadog-specific, json-based logback configuration, and annotate logs accordingly throughout the application

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -85,31 +85,32 @@ scalacOptions ++= Seq(
...
+  val websocket           = "1.1"
+  val web3j               = "4.5.1"
+  val guava               = "28.1-jre"
+  val rabbitmq            = "5.9.0"
+  val akka                = "2.6.13"
+  val akkaRabbitMQ        = "6.0.0"
+  val scalaredis          = "3.30"
+  val ddogAgent           = "0.72.0"
+  val logbackVersion      = "1.2.3"
+  val logbackJsonVersion  = "6.2"
```
Comment (gagbo) [inline build.sbt:113]: I updated logback from 1.1.7 to 1.2.3,  added logbackJsonVersion for logstash encoder.

File: src/main/scala/co/ledger/wallet/daemon/services/AccountSynchronizer.scala [hunk 2]
```diff
@@ -425,7 +533,7 @@ class AccountSynchronizer() extends Actor with ActorLogging {
     val accountUrl: String = s"${accountInfo.poolName}/${accountInfo.walletName}/${account.getIndex}"
     log.debug(s"Trying to synchronize $accountUrl...")
     if (onGoingSyncs.size >= DaemonConfiguration.Synchronization.maxOnGoing) {
-      log.debug(
+      log.warning(
```
Comment (twilgenbus-ledger) [inline src/main/scala/co/ledger/wallet/daemon/services/AccountSynchronizer.scala:536]: I don't think this should be moved to a warning

**Review by twilgenbus-ledger — CHANGES_REQUESTED** [review]

**Comment by twilgenbus-ledger** [comment]
Thanks you, that'll be immensely useful.

File: src/it/resources/application.conf [hunk 3]
```diff
@@ -489,14 +489,13 @@ account-synchronizer-mailbox {
...
 akka {
-    wd-blocking-dispatcher {
-      type = Dispatcher
-      executor = "thread-pool-executor"
-      thread-pool-executor {
-        fixed-pool-size = 16
-      }
-      throughput = 1
-    }
+    log-config-on-start = on
```
[the remark this answers is not in the document]
Comment (gagbo) [reply src/it/resources/application.conf:492]: It’s not used anywhere, it’s an old artifact from @mbasire-ledger changes

File: src/main/resources/application.conf.sample [hunk 4]
```diff
@@ -503,14 +503,7 @@ account-synchronizer-mailbox {
...
 akka {
-    wd-blocking-dispatcher {
-      type = Dispatcher
-      executor = "thread-pool-executor"
-      thread-pool-executor {
-        fixed-pool-size = 16
-      }
-      throughput = 1
-    }
+    log-config-on-start = on
```
[the remark this answers is not in the document]
Comment (gagbo) [reply src/main/resources/application.conf.sample:506]: It’s not used anywhere, it’s an old artifact from @mbasire-ledger changes

File: src/main/scala/co/ledger/wallet/daemon/services/AccountSynchronizer.scala (hunk 2 above)
Comment (gagbo) [reply src/main/scala/co/ledger/wallet/daemon/services/AccountSynchronizer.scala:536]: Sure !

**Review by twilgenbus-ledger — APPROVED** [review]
LGTM!

**Pull request merged**